### PR TITLE
Post Auth Keep me signed in

### PR DESCRIPTION
**Pull request by LazarVuckovic-okta**
## Description:

Introduces a handler for the newly added Post Auth keep me signed in remediation.

Per design, the remediation needs to be handled with a custom look and we need to transform the checkbox into two buttons - one for each state of the checkbox.

This feature is in development and is currently behind a feature flag, as well as an option in the Admin console that is by default set to `false`.

## PR Checklist

- [x] Have you verified the basic functionality for this change?
- [x] Did you add tests, as appropriate, following our [Automated Test guidelines](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/2676497890/Automated+Testing+in+the+Signin+Widget)?
- [x] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?
- [x] Did you verify the change by running [downstream monolith artifact](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/102897979/Sign-in+Widget+Development#Sign-inWidgetDevelopment-Instructionstocreateandrunthedownstreamartifact(d16t))? (Provide link to build below)
- [x] Does this PR include noticeable changes to the UI? (If yes, attach screenshots/video below)

### Issue:

- [OKTA-648819](https://oktainc.atlassian.net/browse/OKTA-648819)

### Screenshots:

Gen 2:

image

Gen 3:

image

### Video:

Gen 2: https://drive.google.com/file/d/1_r8sgcuM3zfMwGCYWmPgOckdG3QcKqzk/view?usp=drive_link
Gen 3: https://drive.google.com/file/d/1oUXACVRi8CE1jtVe4agOsUOFEHLGIPs7/view?usp=drive_link

### Downstream Monolith Build:

https://bacon-go.aue1e.saasure.net/commits?artifact=monolith&page=1&pageSize=6&sha=9c2075d6544587962cc7f067a7f2738d5bde028a&tab=main

### CODE REVIEW

**Comment by jakebacher2-okta** [comment]
I don't know if something has changed, but is the top button supposed to be blue? My info might be outdated, but just wanted to make sure what I'm seeing is the expected UX
https://oktawiki.atlassian.net/wiki/spaces/eng/pages/2827878435/Keep+Me+Signed+In+-+KMSI+-+Eng+Onepager + https://www.figma.com/file/Bqh24qvYWkJqRFcG9niOWI/Keep-Me-Signed-In-Enhancements?type=design&node-id=189-4006&mode=design&t=4c5n3J1UmXOU5VEU-0

Looks like @alexdahl-okta already ❤️ 'ed your PR so maybe that counts as a sign-off?

File: packages/@okta/i18n/src/properties/login_ok_PL.properties [hunk 1]
```diff
@@ -1011,6 +1011,10 @@ oie.selfservice.unlock_user.landing.to.app.signing.in.message = 》Åççöûñ
 oie.selfservice.unlock_user.success.message = 》Ýöû çåñ ļöĝ îñ ûšîñĝ ýöûŕ éẋîšţîñĝ ûšéŕñåɱé åñð þåššŵöŕð· ヸ€홝한Ӝฐโ⾼ئ䀕ヸ€홝한Ӝฐโ《
 oie.selfservice.unlock_user.failed.message = 》Ŵé åŕé ûñåƀļé ţö ûñļöçķ ýöûŕ åççöûñţ åţ ţĥîš ţîɱé، þļéåšé çöñţåçţ ýöûŕ åðɱîñîšţŕåţöŕ ヸ€홝한Ӝฐโ⾼ئ䀕ヸ€홝한Ӝฐโ《
 oie.selfservice.user.unlock.not.allowed = 》Šéļƒ Šéŕṽîçé Ûñļöçķ îš ñöţ åļļöŵéð åţ ţĥîš ţîɱé· Þļéåšé çöñţåçţ šûþþöŕţ ƒöŕ åššîšţåñçé· ヸ€홝한Ӝฐโ⾼ئ䀕ヸ€홝한Ӝฐโ《
+oie.kmsi.title = 》Ķééþ ɱé šîĝñéð îñ 한Ӝฐโ⾼ئ䀕ヸ€홝한Ӝฐโ《
+oie.kmsi.subtitle = 》Šîĝñ îñ åñð éñţéŕ šéçûŕîţý ɱéţĥöðš ļéšš ƒŕéǫûéñţļý· 홝한Ӝฐโ⾼ئ䀕ヸ€홝한Ӝฐโ《
+oie.kmsi.accept = 》Šţåý šîĝñéð îñ โ⾼ئ䀕ヸ€홝한Ӝฐโ《
+oie.kmsi.reject = 》Ðöñ´ţ šţåý šîĝñéð îñ €홝한Ӝฐโ⾼ئ䀕ヸ€홝한Ӝฐโ《
```
Comment (jakebacher2-okta) [inline packages/@okta/i18n/src/properties/login_ok_PL.properties:1017]: do we merge in ok_PL files? I thought they were auto-generated? Maybe not for the SIW.

File: packages/@okta/i18n/src/properties/login_ok_SK.properties [hunk 2]
```diff
@@ -1011,6 +1011,10 @@ oie.selfservice.unlock_user.landing.to.app.signing.in.message = [[okta-signin-wi
 oie.selfservice.unlock_user.success.message = [[okta-signin-widget:login: oie.selfservice.unlock_user.success.message]]
 oie.selfservice.unlock_user.failed.message = [[okta-signin-widget:login: oie.selfservice.unlock_user.failed.message]]
 oie.selfservice.user.unlock.not.allowed = [[okta-signin-widget:login: oie.selfservice.user.unlock.not.allowed]]
+oie.kmsi.title = [[okta-signin-widget:login: oie.kmsi.title]]
+oie.kmsi.subtitle = [[okta-signin-widget:login: oie.kmsi.subtitle]]
+oie.kmsi.accept = [[okta-signin-widget:login: oie.kmsi.accept]]
+oie.kmsi.reject = [[okta-signin-widget:login: oie.kmsi.reject]]
```
Comment (jakebacher2-okta) [inline packages/@okta/i18n/src/properties/login_ok_SK.properties:1017]: Same comment as with the ok_PL file

**Review by jakebacher2-okta — COMMENTED** [review]
Couple comments. If the API call were to fail for whatever reason, do you have an idea of what it looks like? Is it a generic error message?

**Comment by alexdahl-okta** [comment]
> is the top button supposed to be blue?

Ah yeah, we changed both buttons to secondary (gray). The decision affects the user's safety and depends on context we don't know about, so we shouldn't recommend a choice.

https://www.figma.com/file/Bqh24qvYWkJqRFcG9niOWI/Keep-Me-Signed-In-Enhancements?type=design&node-id=1665%3A4139&mode=design&t=Z7xIUHdAvC0iEDPT-1

**Comment by LazarVuckovic-okta** [comment]
> Couple comments. If the API call were to fail for whatever reason, do you have an idea of what it looks like? Is it a generic error message?

@jakebacher2-okta I forced some errors in debug mode, here are the screenshots

Screenshot 2023-11-07 at 12 09 38 PM
Screenshot 2023-11-07 at 12 05 29 PM

File: packages/@okta/i18n/src/properties/login_ok_PL.properties (hunk 1 above)
Comment (LazarVuckovic-okta) [reply packages/@okta/i18n/src/properties/login_ok_PL.properties:1017]: yeah, we commit these changes in SIW

File: packages/@okta/i18n/src/properties/login_ok_SK.properties (hunk 2 above)
Comment (LazarVuckovic-okta) [reply packages/@okta/i18n/src/properties/login_ok_SK.properties:1017]: yeah, we commit these changes in SIW

File: src/v3/src/transformer/layout/keepMeSignedIn/transformKeepMeSignedIn.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      content: loc('oie.kmsi.title', 'login'),
+    },
+  };
+
+  const descriptionElement: DescriptionElement = {
+    type: 'Description',
+    options: {
+      content: loc('oie.kmsi.subtitle', 'login'),
+    },
+  };
```
[the remark this answers is not in the document]
Comment (LazarVuckovic-okta) [reply src/v3/src/transformer/layout/keepMeSignedIn/transformKeepMeSignedIn.ts]: Thank you, updated this and the snapshot

**Review by jakebacher2-okta — APPROVED** [review]
LGTM, please get SIW home team approval

File: src/v3/src/transformer/testAttribute/transform.ts [hunk 4]
```diff
@@ -81,7 +81,7 @@ const updateRegularButtons: TransformStepFn = (formbag) => {
     },
     callback: (element) => {
       const { options } = element as ButtonElement | LinkElement;
-      if (options.step) {
+      if (options.step && map[options.step]) {
```
Comment (LazarVuckovic-okta) [inline src/v3/src/transformer/testAttribute/transform.ts:84]: Fixes the issue with not being able to pass `dataSe` if the button had a `step` defined

File: src/v3/test/integration/__snapshots__/authenticator-piv-cac-verification.test.tsx.snap [hunk 5]
```diff
@@ -61,6 +61,7 @@ exports[`authenticator-piv-cac-verification renders form with custom PIV/CAC But
                   >
                     <button
                       class="MuiButtonBase-root MuiButton-root MuiButton-secondary MuiButton-secondaryPrimary MuiButton-sizeMedium MuiButton-secondarySizeMedium MuiButton-disableElevation MuiButton-fullWidth MuiButton-root MuiButton-secondary MuiButton-secondaryPrimary MuiButton-sizeMedium MuiButton-secondarySizeMedium MuiButton-disableElevation MuiButton-fullWidth custom-piv-class piv-button emotion-14"
+                      data-se="piv-card-button"
```
Comment (LazarVuckovic-okta) [inline src/v3/test/integration/__snapshots__/authenticator-piv-cac-verification.test.tsx.snap:64]: Updated after the data-se change in the `testAttribute` transform

File: src/v3/test/integration/__snapshots__/authenticator-piv-cac-verification.test.tsx.snap [hunk 6]
```diff
@@ -380,6 +381,7 @@ exports[`authenticator-piv-cac-verification renders form with idpDisplay as PRIM
                   >
                     <button
                       class="MuiButtonBase-root MuiButton-root MuiButton-secondary MuiButton-secondaryPrimary MuiButton-sizeMedium MuiButton-secondarySizeMedium MuiButton-disableElevation MuiButton-fullWidth MuiButton-root MuiButton-secondary MuiButton-secondaryPrimary MuiButton-sizeMedium MuiButton-secondarySizeMedium MuiButton-disableElevation MuiButton-fullWidth  piv-button emotion-14"
+                      data-se="piv-card-button"
```
Comment (LazarVuckovic-okta) [inline src/v3/test/integration/__snapshots__/authenticator-piv-cac-verification.test.tsx.snap:384]: Updated after the data-se change in the `testAttribute` transform

File: src/v3/test/integration/__snapshots__/authenticator-piv-cac-verification.test.tsx.snap [hunk 7]
```diff
@@ -857,6 +859,7 @@ exports[`authenticator-piv-cac-verification renders form with idpDisplay as SECO
                   >
                     <button
                       class="MuiButtonBase-root MuiButton-root MuiButton-secondary MuiButton-secondaryPrimary MuiButton-sizeMedium MuiButton-secondarySizeMedium MuiButton-disableElevation MuiButton-fullWidth MuiButton-root MuiButton-secondary MuiButton-secondaryPrimary MuiButton-sizeMedium MuiButton-secondarySizeMedium MuiButton-disableElevation MuiButton-fullWidth  piv-button emotion-39"
+                      data-se="piv-card-button"
```
Comment (LazarVuckovic-okta) [inline src/v3/test/integration/__snapshots__/authenticator-piv-cac-verification.test.tsx.snap:862]: Updated after the data-se change in the `testAttribute` transform

**Review by lesterchoi-okta — APPROVED** [review]

**Pull request merged**